backend/src/http/assets.rs

```diff
@@ -288,14 +308,47 @@ fn frontend_config(config: &Config) -> serde_json::Value {
             "studioUrl": config.opencast.studio_url().to_string(),
             "editorUrl": config.opencast.editor_url().to_string(),
         },
-        "logo": {
-            "large": logo_obj(&config.theme.logo.large),
-            "small": config.theme.logo.small.as_ref().map(logo_obj),
-            "largeDark": config.theme.logo.large_dark.as_ref().map(logo_obj),
-            "smallDark": config.theme.logo.small_dark.as_ref().map(logo_obj),
-        },
+        "logos": logo_entries,
         "sync": {
             "pollPeriod": config.sync.poll_period.as_secs_f64(),
         },
     })
 }
+
+/// Generates HTTP path for a logo based on its `size`, `mode` and `lang` attributes.
+/// These are joined with `-`.
+/// Defaults to `"default-logo"` if all attributes are wildcards (`"*"`).
+fn generate_http_path(logo: &LogoDef) -> String {
+    let (size, mode, lang) = get_logo_key(logo);
+    let path_parts = [size, mode, lang]
+        .into_iter()
+        .filter(|part| part != "*")
+        .collect::<Vec<_>>();
+
+    if path_parts.is_empty() {
+        "logo-default.svg".to_string()
+    } else {
+        format!("logo-{}.svg", path_parts.join("-"))
+    }
```

Comment:
Alternative suggestion:

```rs
let [size, mode, lang] = [logo.size, logo.mode, logo.lang]
    .map(|opt| opt.map(|v| format!("-{v}")).unwrap_or_default());

format!("logo{size}{mode}{lang}.svg")
```

It's slightly different: if all fields are `None`, it results in `logo.svg` instead of `logo-default.svg`. But it is shorter and avoids using `get_logo_key` (which I think is the last usage of it, so it can be removed then).